### DISCUSSION TITLE
Auth SSR and Supabase UI

### DISCUSSION THREAD

**Discussion by Etheonor** [supabase/supabase#1188] [Questions]
Hello!

One major problem I have today with Supabase Auth is that it save the session on localstorage and not a cookie. 

I found this solution here : https://github.com/supabase/gotrue-js/pull/33 but I don't know how to make it compatible with Supabase UI. 

Right now I'm user the Auth.UsercontextProvider HOC on my NextJS application. 

Is there any way to use this HOC with the cookie method, or am I condemned to choose between SSR and Supabase UI?

**Comment by kiwicopple** [comment]
Hi @Etheonor . Does this example help? https://github.com/supabase/supabase/tree/master/examples/nextjs-with-supabase-auth

**Reply by Etheonor** [reply]
Hello @kiwicopple and thank you for the link!

If I understand correctly, the useEffect on the index page is where the cookie is set up. What happen if I disconnect on another page? Should I make another HOC with just this useEffect and wrap my entire app around it? Is that something viable?

**Reply by kiwicopple** [reply]
I'm no too sure what you mean by "disconnect", so feel free to dismiss this if I've misunderstood:

Once a cookie is set in the browser, it will always be passed as a header (automatically) to the server. So you will need to always "inspect" the `req` headers on the server to figure out if there is a cookie

I don't know the details of you HOC but if you are setting a cookie in the browser, the HOC should be able to check if the cookie exists too.

**Reply by Etheonor** [reply]
Ok I didn't understand the implementation correctly ^^ I managed to make it work. 

I changed a bit of code because the implementation on your github didn't seems to work on my side : 

The event USER_UPDATED doesn't trigger anything. I replaced it by if ((event === "SIGNED_OUT") | (event === "SIGNED_IN")) and the cookie is created/deleted when I log in / log out.

**Reply by Etheonor** [reply]
Ok well it works, but with some issues. 

I already opened a topic about the refresh token, it's still a problem. https://github.com/supabase/supabase/discussions/740

After an hour, if I try to do an action with the token from the cookie, I got an error message JWT expired. But there is no way to handle it. Your example nextjs-with-supabase-auth should also have this problem.

**Reply by Nick-Mazuk** [reply]
Running into the JWT expiration issue myself.  Any updates on how to solve that? Didn't see anything in #740.

**Reply by Etheonor** [reply]
Update the npm package to the latest supabase ui version fixed the issue for me. I saw a merge request about that a few weeks ago
